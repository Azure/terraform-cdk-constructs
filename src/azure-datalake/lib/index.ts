export * from "./filesystem";
export * from "./path";
export * from "./datalake";
