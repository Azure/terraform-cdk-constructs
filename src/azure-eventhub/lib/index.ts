export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
