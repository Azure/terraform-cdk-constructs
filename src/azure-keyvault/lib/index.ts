export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

