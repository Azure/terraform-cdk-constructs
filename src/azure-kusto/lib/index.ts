export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
