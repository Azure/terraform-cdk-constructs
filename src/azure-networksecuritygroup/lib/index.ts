export * from './preconfigured-rules';
export * from './securitygroup';
