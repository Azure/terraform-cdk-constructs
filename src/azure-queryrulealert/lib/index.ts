export * from './query-rule-alert'