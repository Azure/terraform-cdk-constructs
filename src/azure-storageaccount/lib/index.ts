export * from './account';
export * from './container';
export * from './fileshare';
export * from './queue';
export * from './table';
