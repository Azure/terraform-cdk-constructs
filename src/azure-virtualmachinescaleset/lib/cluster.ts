import {
  LinuxVirtualMachineSourceImageReference,
  LinuxVirtualMachineOsDisk,
  LinuxVirtualMachineAdminSshKey,
  LinuxVirtualMachineIdentity,
} from "@cdktf/provider-azurerm/lib/linux-virtual-machine";
import {
  LinuxVirtualMachineScaleSet,
  LinuxVirtualMachineScaleSetNetworkInterfaceIpConfigurationPublicIpAddress,
} from "@cdktf/provider-azurerm/lib/linux-virtual-machine-scale-set";
import { Subnet } from "@cdktf/provider-azurerm/lib/subnet";
import { VirtualMachineScaleSetExtensionA } from "@cdktf/provider-azurerm/lib/virtual-machine-scale-set-extension";
import {
  WindowsVirtualMachineOsDisk,
  WindowsVirtualMachineSourceImageReference,
} from "@cdktf/provider-azurerm/lib/windows-virtual-machine";
import {
  WindowsVirtualMachineScaleSet,
  WindowsVirtualMachineScaleSetNetworkInterfaceIpConfigurationPublicIpAddress,
} from "@cdktf/provider-azurerm/lib/windows-virtual-machine-scale-set";
import * as cdktf from "cdktf";
import { Construct } from "constructs";
import {
  WindowsImageReferences,
  LinuxImageReferences,
} from "../../azure-virtualmachine";
import { Network } from "../../azure-virtualnetwork/lib/network";
import { AzureResource } from "../../core-azure/lib";

export interface LinuxClusterProps {
  /**
   * The Azure location where the virtual machine scale set should be created.
   * @default "eastus"
   */
  readonly location?: string;

  /**
   * The name of the virtual machine scale set.
   * @default - Uses the name derived from the construct path.
   */
  readonly name?: string;

  /**
   * The admin username for the virtual machine.
   */
  readonly adminUsername?: string;

  /**
   * The admin password for the virtual machine.
   */
  readonly adminPassword?: string;

  /**
   * The name of the resource group in which the virtual machine scale set will be created.
   */
  readonly resourceGroupName: string;

  /**
   * The size of the virtual machines in the scale set.
   * @default "Standard_B2s"
   */
  readonly sku?: string;

  /**
   * Custom data to pass to the virtual machines upon creation.
   */
  readonly userData?: string;

  /**
   * An array of SSH keys for the admin user.
   */
  readonly adminSshKey?: LinuxVirtualMachineAdminSshKey[] | cdktf.IResolvable;

  /**
   * The availability zone(s) in which the VMs should be placed.
   */
  readonly zones?: string[];

  /**
   * Managed identity settings for the VMs.
   */
  readonly identity?: LinuxVirtualMachineIdentity;

  /**
   * The source image reference for the virtual machines.
   * @default - Uses a default Ubuntu image.
   */
  readonly sourceImageReference?: LinuxVirtualMachineSourceImageReference;

  /**
   * The ID of the source image for the virtual machines.
   */
  readonly sourceImageId?: string;

  /**
   * Tags to apply to the virtual machine scale set.
   */
  readonly tags?: { [key: string]: string };

  /**
   * The OS disk configuration for the virtual machines.
   * @default - Uses a disk with caching set to "ReadWrite" and storage account type "Standard_LRS".
   */
  readonly osDisk?: LinuxVirtualMachineOsDisk;

  /**
   * The subnet in which the virtual machines will be placed.
   */
  readonly subnet?: Subnet;

  /**
   * The allocation method for the public IP.
   */
  readonly publicIPAddress?: LinuxVirtualMachineScaleSetNetworkInterfaceIpConfigurationPublicIpAddress[];

  /**
   * Custom data to pass to the virtual machines upon creation.
   */
  readonly customData?: string;

  /**
   * The number of VM instances in the scale set.
   * @default 2
   */
  readonly instances?: number;

  /**
   * Specifies the scale set's upgrade policy settings.
   */
  readonly upgradePolicyMode?: string;

  /**
   * Specifies if the VMSS should be overprovisioned.
   * @default true
   */
  readonly overprovision?: boolean;

  /**
   * Specifies the scale-in policy for the VMSS.
   */
  readonly scaleInPolicy?: string;

  /**
   * Boot diagnostics settings for the VMSS.
   */
  readonly bootDiagnosticsStorageURI?: string;

  /**
   * Enable SSH Azure AD Login, required managed identity to be set.
   * @default false
   */
  readonly enableSshAzureADLogin?: boolean;
}

export class LinuxCluster extends AzureResource {
  public readonly props: LinuxClusterProps;
  public readonly resourceGroupName: string;
  public readonly id: string;
  public readonly name: string;
  public readonly fqn: string;

  /**
   * Constructs a new instance of the AzureLinuxVirtualMachine class.
   *
   * @param scope - The scope in which this construct is defined.
   * @param id - The ID of this construct.
   * @param props - The properties for defining a Linux Virtual Machine.
   */

  constructor(scope: Construct, id: string, props: LinuxClusterProps) {
    super(scope, id);

    this.props = props;
    this.resourceGroupName = props.resourceGroupName;

    const pathName = this.node.path.split("/")[0];

    const defaults = {
      name: props.name || pathName,
      adminUsername: props.adminUsername || `admin${pathName}`,
      location: props.location || "eastus",
      sku: props.sku || "Standard_B2s",
      instances: props.instances || 1,
      osDisk: props.osDisk || {
        caching: "ReadWrite",
        storageAccountType: "Standard_LRS",
      },
      sourceImageReference:
        props.sourceImageReference || LinuxImageReferences.UbuntuServer2204LTS,
      subnet:
        props.subnet ||
        new Network(this, "vnet", {
          resourceGroupName: props.resourceGroupName,
        }).subnets.default,
    };

    const azurermLinuxVirtualMachineScaleSet = new LinuxVirtualMachineScaleSet(
      this,
      "vmss",
      {
        ...defaults,
        resourceGroupName: props.resourceGroupName,
        adminPassword: props.adminPassword,
        disablePasswordAuthentication: props.adminPassword ? false : true,
        tags: props.tags,
        networkInterface: [
          {
            name: `nic-${defaults.name}`,
            primary: true,
            ipConfiguration: [
              {
                name: "internal",
                subnetId: defaults.subnet.id,
                primary: true,
                publicIpAddress: props.publicIPAddress,
              },
            ],
          },
        ],
        osDisk: {
          ...defaults.osDisk,
        },
        sourceImageId: props.sourceImageId,
        customData: props.customData
          ? Buffer.from(props.customData).toString("base64")
          : undefined,
        userData: props.userData
          ? Buffer.from(props.userData).toString("base64")
          : undefined,
        adminSshKey: props.adminSshKey,
        identity: props.identity,
      },
    );

    this.id = azurermLinuxVirtualMachineScaleSet.id;
    this.name = azurermLinuxVirtualMachineScaleSet.name;
    this.fqn = azurermLinuxVirtualMachineScaleSet.fqn;

    // Enable SSH Azure AD Login if specified
    if (props.enableSshAzureADLogin) {
      new VirtualMachineScaleSetExtensionA(this, "AADSSHlogin", {
        name: "AADSSHLoginForLinux",
        virtualMachineScaleSetId: this.id,
        publisher: "Microsoft.Azure.ActiveDirectory",
        type: "AADSSHLoginForLinux",
        typeHandlerVersion: "1.0",
      });
    }
  }
}

export interface WindowsClusterProps {
  /**
   * The Azure location where the virtual machine should be created.
   * @default "eastus"
   */
  readonly location?: string;

  /**
   * The name of the virtual machine.
   * @default - Uses the name derived from the construct path.
   */
  readonly name?: string;

  /**
   * The name of the resource group in which the virtual machine will be created.
   */
  readonly resourceGroupName: string;

  /**
   * The size of the virtual machine.
   * @default "Standard_B2s"
   */
  readonly sku?: string;

  /**
   * The admin username for the virtual machine.
   */
  readonly adminUsername: string;

  /**
   * The admin password for the virtual machine.
   */
  readonly adminPassword: string;

  /**
   * The availability zone(s) in which the VMs should be placed.
   */
  readonly zones?: string[];

  /**
   * The number of VM instances in the scale set.
   * @default 2
   */
  readonly instances?: number;

  /**
   * The source image reference for the virtual machine.
   * @default - Uses WindowsServer2022DatacenterCore.
   */
  readonly sourceImageReference?: WindowsVirtualMachineSourceImageReference;

  /**
   * The ID of the source image for the virtual machine.
   */
  readonly sourceImageId?: string;

  /**
   * Tags to apply to the virtual machine.
   */
  readonly tags?: { [key: string]: string };

  /**
   * The OS disk configuration for the virtual machine.
   * @default - Uses a disk with caching set to "ReadWrite" and storage account type "Standard_LRS".
   */
  readonly osDisk?: WindowsVirtualMachineOsDisk;

  /**
   * The subnet in which the virtual machine will be placed.
   * @default - Uses the default subnet from a new virtual network.
   */
  readonly subnet?: Subnet;

  /**
   * The allocation method for the public IP.
   */
  readonly publicIPAddress?: WindowsVirtualMachineScaleSetNetworkInterfaceIpConfigurationPublicIpAddress[];

  /**
   * Specifies the scale set's upgrade policy settings.
   */
  readonly upgradePolicyMode?: string;

  /**
   * Custom data to pass to the virtual machine upon creation.
   */
  readonly customData?: string;

  /**
   * Custom data to bootstrap the virtual machine. Automatically triggers Azure Custom Script extension to deploy code in custom data.
   */
  readonly boostrapCustomData?: string;

  /**
   * Bootdiagnostics settings for the VM.
   */
  readonly bootDiagnosticsStorageURI?: string;

  /**
   * Specifies if the VMSS should be overprovisioned.
   * @default true
   */
  readonly overprovision?: boolean;

  /**
   * Specifies the scale-in policy for the VMSS.
   */
  readonly scaleInPolicy?: string;
}

export class WindowsCluster extends AzureResource {
  readonly props: WindowsClusterProps;
  public readonly resourceGroupName: string;
  public readonly id: string;
  public readonly name: string;

  /**
   * Constructs a new instance of the AzureWindowsVirtualMachine class.
   *
   * @param scope - The scope in which this construct is defined.
   * @param id - The ID of this construct.
   * @param props - The properties for defining a Windows Virtual Machine.
   */
  constructor(scope: Construct, id: string, props: WindowsClusterProps) {
    super(scope, id);

    this.props = props;
    this.resourceGroupName = props.resourceGroupName;

    const pathName = this.node.path.split("/")[0];

    // Default configurations for the virtual machine.
    const defaults = {
      name: props.name || this.node.path.split("/")[0],
      adminUsername: props.adminUsername || `admin${pathName}`,
      location: props.location || "eastus",
      sku: props.sku || "Standard_B2s",
      instances: props.instances || 1,
      osDisk: props.osDisk || {
        caching: "ReadWrite",
        storageAccountType: "Standard_LRS",
      },
      sourceImageReference:
        props.sourceImageReference ||
        WindowsImageReferences.WindowsServer2022DatacenterCore,
      subnet:
        props.subnet ||
        new Network(this, "vnet", {
          resourceGroupName: props.resourceGroupName,
        }).subnets.default,
    };

    // Base64 encode custom data if provided.
    const customData = props.customData || props.boostrapCustomData;
    const base64CustomData = customData
      ? Buffer.from(customData).toString("base64")
      : undefined;

    // Create the Windows Virtual Machine.
    const azurermWindowsVirtualMachine = new WindowsVirtualMachineScaleSet(
      this,
      "vmss",
      {
        ...defaults,
        resourceGroupName: props.resourceGroupName,
        adminUsername: props.adminUsername,
        adminPassword: props.adminPassword,
        tags: props.tags,
        networkInterface: [
          {
            name: `nic-${defaults.name}`,
            primary: true,
            ipConfiguration: [
              {
                name: "internal",
                subnetId: defaults.subnet.id,
                primary: true,
                publicIpAddress: props.publicIPAddress,
              },
            ],
          },
        ],
        osDisk: {
          ...defaults.osDisk,
        },
        sourceImageId: props.sourceImageId,
        customData: base64CustomData,
        bootDiagnostics: { storageAccountUri: props.bootDiagnosticsStorageURI },
      },
    );

    this.id = azurermWindowsVirtualMachine.id;
    this.name = azurermWindowsVirtualMachine.name;

    // Bootstrap VM with custom script extension if bootstrap custom data is provided.
    if (props.boostrapCustomData) {
      new VirtualMachineScaleSetExtensionA(this, "script-ext", {
        name: `${this.name}-script-ext`,
        virtualMachineScaleSetId: this.id,
        publisher: "Microsoft.Compute",
        type: "CustomScriptExtension",
        typeHandlerVersion: "1.10",
        protectedSettings:
          '{"commandToExecute": "rename  C:\\\\AzureData\\\\CustomData.bin  postdeploy.ps1 & powershell -ExecutionPolicy Unrestricted -File C:\\\\AzureData\\\\postdeploy.ps1"}',
      });
    }
  }
}
