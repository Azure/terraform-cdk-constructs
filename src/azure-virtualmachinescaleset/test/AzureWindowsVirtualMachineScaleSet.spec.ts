import { AzurermProvider } from "@cdktf/provider-azurerm/lib/provider";
import { Testing, TerraformStack } from "cdktf";
import "cdktf/lib/testing/adapters/jest";
import * as vmss from "..";
import { exampleAzureWindowsVirtualMachineScaleSet } from "../test/ExampleAzureWindowsVirtualMachineScaleSet";

describe("Azure Windows Virtual Machine Scale Set With Defaults", () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "testAzureVMSSWithDefaults");

    new AzurermProvider(stack, "azureFeature", { features: {} });

    new vmss.WindowsCluster(stack, "testVirtualMachineScaleSet", {
      resourceGroupName: "testResourceGroup",
      adminUsername: "testAdmin",
      adminPassword: "testPassword&@34$$123",
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Windows Virtual Machine  Scale Set with defaults and checks snapshot", () => {
    expect(Testing.synth(stack)).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    expect(fullSynthResult).toPlanSuccessfully(); // Use the saved result
  });
});

describe("Azure Windows Virtual Machine  Scale Set Example", () => {
  it("renders the Azure Windows Virtual Machine Scale Set and checks snapshot", () => {
    expect(
      Testing.synth(
        new exampleAzureWindowsVirtualMachineScaleSet(
          Testing.app(),
          "testAzureWindowsVirtualMachineScaleSetExample",
        ),
      ),
    ).toMatchSnapshot();
  });

  it("check if the produced terraform configuration is valid", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(
      Testing.fullSynth(
        new exampleAzureWindowsVirtualMachineScaleSet(
          Testing.app(),
          "testAzureWindowsVirtualMachineScaleSetExample",
        ),
      ),
    ).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    // We need to do a full synth to plan the terraform configuration
    expect(
      Testing.fullSynth(
        new exampleAzureWindowsVirtualMachineScaleSet(
          Testing.app(),
          "testAzureWindowsVirtualMachineScaleSetExample",
        ),
      ),
    ).toPlanSuccessfully();
  });
});
