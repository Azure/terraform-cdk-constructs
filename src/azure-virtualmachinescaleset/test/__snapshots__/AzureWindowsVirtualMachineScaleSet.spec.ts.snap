// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Windows Virtual Machine  Scale Set Example renders the Azure Windows Virtual Machine Scale Set and checks snapshot 1`] = `
"{
  "data": {
    "azurerm_client_config": {
      "CurrentClientConfig": {
      }
    },
    "azurerm_monitor_diagnostic_categories": {
      "vmss_diagsettings_diagcategories_E03AFDA3": {
        "resource_id": "\${azurerm_windows_virtual_machine_scale_set.vmss_EA5696D9.id}"
      }
    }
  },
  "output": {
    "public_ip_name": {
      "value": "testpublicip"
    },
    "resource_group_name": {
      "value": "\${azurerm_resource_group.rg.name}"
    },
    "vm_name": {
      "value": "\${azurerm_windows_virtual_machine_scale_set.vmss_EA5696D9.name}"
    },
    "vm_size": {
      "value": "Standard_B1s"
    }
  },
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_monitor_diagnostic_setting": {
      "vmss_diagsettings_22C55F67": {
        "dynamic": {
          "enabled_log": {
            "content": {
              "category": "\${enabled_log.value}"
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.vmss_diagsettings_diagcategories_E03AFDA3.log_category_types)}"
          },
          "metric": {
            "content": {
              "category": "\${metric.value}"
            },
            "for_each": "\${tolist(data.azurerm_monitor_diagnostic_categories.vmss_diagsettings_diagcategories_E03AFDA3.metrics)}"
          }
        },
        "name": "diagsettings",
        "storage_account_id": "\${azurerm_storage_account.storage.id}",
        "target_resource_id": "\${azurerm_windows_virtual_machine_scale_set.vmss_EA5696D9.id}"
      }
    },
    "azurerm_resource_group": {
      "rg": {
        "location": "eastus",
        "name": "rg-\${var.name}"
      }
    },
    "azurerm_role_assignment": {
      "vmss_TfTokenTOKEN63Contributor_role_5A791E98": {
        "principal_id": "\${data.azurerm_client_config.CurrentClientConfig.object_id}",
        "role_definition_name": "Contributor",
        "scope": "\${azurerm_windows_virtual_machine_scale_set.vmss_EA5696D9.id}"
      }
    },
    "azurerm_storage_account": {
      "storage": {
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "location": "\${azurerm_resource_group.rg.location}",
        "min_tls_version": "TLS1_2",
        "name": "sta\${var.name}88s96",
        "network_rules": {
          "bypass": [
            "AzureServices"
          ],
          "default_action": "Deny"
        },
        "public_network_access_enabled": false,
        "resource_group_name": "\${azurerm_resource_group.rg.name}"
      }
    },
    "azurerm_subnet": {
      "subnet1": {
        "address_prefixes": [
          "10.0.1.0/24"
        ],
        "name": "subnet1",
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "virtual_network_name": "\${azurerm_virtual_network.vnet.name}"
      }
    },
    "azurerm_virtual_machine_scale_set_extension": {
      "vmss_script-ext_0F9B18A4": {
        "name": "\${azurerm_windows_virtual_machine_scale_set.vmss_EA5696D9.name}-script-ext",
        "protected_settings": "{\\"commandToExecute\\": \\"rename  C:\\\\\\\\AzureData\\\\\\\\CustomData.bin  postdeploy.ps1 & powershell -ExecutionPolicy Unrestricted -File C:\\\\\\\\AzureData\\\\\\\\postdeploy.ps1\\"}",
        "publisher": "Microsoft.Compute",
        "type": "CustomScriptExtension",
        "type_handler_version": "1.10",
        "virtual_machine_scale_set_id": "\${azurerm_windows_virtual_machine_scale_set.vmss_EA5696D9.id}"
      }
    },
    "azurerm_virtual_network": {
      "vnet": {
        "address_space": [
          "10.0.0.0/16"
        ],
        "location": "\${azurerm_resource_group.rg.location}",
        "name": "vnet-\${var.name}",
        "resource_group_name": "\${azurerm_resource_group.rg.name}"
      }
    },
    "azurerm_windows_virtual_machine_scale_set": {
      "vmss_EA5696D9": {
        "admin_password": "Password1234!",
        "admin_username": "testadmin",
        "boot_diagnostics": {
          "storage_account_uri": "\${azurerm_storage_account.storage.primary_blob_endpoint}"
        },
        "custom_data": "SW5zdGFsbC1XaW5kb3dzRmVhdHVyZSAtTmFtZSBXZWItU2VydmVyOyAkd2Vic2l0ZSA9ICc8aDE+SGVsbG8gV29ybGQhPC9oMT4nOyBTZXQtQ29udGVudCAiQzpcaW5ldHB1Ylx3d3dyb290XGlpc3N0YXJ0Lmh0bSIgJHdlYnNpdGU=",
        "instances": 1,
        "location": "eastus",
        "name": "\${var.name}",
        "network_interface": [
          {
            "ip_configuration": [
              {
                "name": "internal",
                "primary": true,
                "public_ip_address": [
                  {
                    "name": "testpublicip"
                  }
                ],
                "subnet_id": "\${azurerm_subnet.subnet1.id}"
              }
            ],
            "name": "nic-\${var.name}",
            "primary": true
          }
        ],
        "os_disk": {
          "caching": "ReadWrite",
          "storage_account_type": "Standard_LRS"
        },
        "resource_group_name": "\${azurerm_resource_group.rg.name}",
        "sku": "Standard_B1s",
        "source_image_reference": {
          "offer": "WindowsServer",
          "publisher": "MicrosoftWindowsServer",
          "sku": "2019-Datacenter",
          "version": "latest"
        }
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.70.0"
      }
    }
  },
  "variable": {
    "name": {
      "default": "test",
      "description": "System name used to randomize the resources",
      "type": "string"
    }
  }
}"
`;

exports[`Azure Windows Virtual Machine Scale Set With Defaults renders an Azure Windows Virtual Machine  Scale Set with defaults and checks snapshot 1`] = `
"{
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_resource_group": {
      "testVirtualMachineScaleSet_rg_41B4540E": {
        "location": "eastus",
        "name": "rg-undefined"
      }
    },
    "azurerm_subnet": {
      "testVirtualMachineScaleSet_vnet_default_097110D2": {
        "address_prefixes": [
          "10.1.0.0/24"
        ],
        "name": "default",
        "resource_group_name": "\${azurerm_resource_group.testVirtualMachineScaleSet_rg_41B4540E.name}",
        "virtual_network_name": "\${azurerm_virtual_network.testVirtualMachineScaleSet_vnet_6B0AF69B.name}"
      }
    },
    "azurerm_virtual_network": {
      "testVirtualMachineScaleSet_vnet_6B0AF69B": {
        "address_space": [
          "10.0.0.0/16"
        ],
        "location": "eastus",
        "name": "vnet-testAzureVMSSWithDefaults",
        "resource_group_name": "\${azurerm_resource_group.testVirtualMachineScaleSet_rg_41B4540E.name}"
      }
    },
    "azurerm_windows_virtual_machine_scale_set": {
      "testVirtualMachineScaleSet_vmss_4AC909AF": {
        "admin_password": "testPassword&@34$$123",
        "admin_username": "testAdmin",
        "instances": 1,
        "location": "eastus",
        "name": "testAzureVMSSWithDefaults",
        "network_interface": [
          {
            "ip_configuration": [
              {
                "name": "internal",
                "primary": true,
                "subnet_id": "\${azurerm_subnet.testVirtualMachineScaleSet_vnet_default_097110D2.id}"
              }
            ],
            "name": "nic-testAzureVMSSWithDefaults",
            "primary": true
          }
        ],
        "os_disk": {
          "caching": "ReadWrite",
          "storage_account_type": "Standard_LRS"
        },
        "resource_group_name": "\${azurerm_resource_group.testVirtualMachineScaleSet_rg_41B4540E.name}",
        "sku": "Standard_B2s",
        "source_image_reference": {
          "offer": "WindowsServer",
          "publisher": "MicrosoftWindowsServer",
          "sku": "2022-Datacenter-Core",
          "version": "latest"
        }
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.70.0"
      }
    }
  }
}"
`;
